eodatasets3/model.py

```diff
@@ -373,6 +372,75 @@ def producer_abbreviated(self) -> Optional[str]:
             )
 
 
+class ComplicatedNamingConventionsDerivatives(ComplicatedNamingConventions):
+    """
+    This class is inherited from ComplicatedNamingConventions
+    and overrides few attributes specific to C3 data processing for data other than ARD.
+    """
+
+    @classmethod
+    def for_c3_derivatives(cls, dataset: EoFields, uri=DEA_URI_PREFIX):
+        """
+        The required fields for the c3 data processing are controlled here.
+        """
+        return cls(
+            dataset=dataset,
+            base_product_uri=uri,
+            required_fields=(
+                "eo:platform",
+                "odc:dataset_version",
+                "odc:collection_number",
+                "odc:processing_datetime",
+                "odc:producer",
+                "odc:product_family",
+                "odc:region_code",
+                "dea:dataset_maturity",
+            ),
+        )
+
+    @property
+    def _org_collection_number(self) -> Optional[int]:
+        # Deliberately fail if collection_number is not defined in the config yaml
+        return int(self.dataset.collection_number)
+
+    def _product_group(self, subname=None) -> str:
+        # Computues product group, e.g "ga_ls_wo_3"
+        # Deliberately fail if any of these attributes not found.
+        parts = [
+            self.producer_abbreviated,
+            self.platform_abbreviated,
+            self.dataset.product_family,
+        ]
+        return "_".join(parts)
+
+    def destination_folder(self, base: Path):
+        self._check_enough_properties_to_name()
+        parts = [self.product_name, self.dataset.dataset_version.replace(".", "-")]
+        parts.extend(utils.subfolderise(self.dataset.region_code))
+        parts.extend(f"{self.dataset.datetime:%Y/%m/%d}".split("/"))
+        return base.joinpath(*parts)
+
+    def _dataset_label(self, sub_name: str = None):
+        """
+        Responsible for producing the string of product name, regioncode, datetime and maturity
+        ex: 'ga_ls_wo_3_090081_1998-07-30_interim'
+
+        Redundant parameter sub_name is required, since the parent class and other invocations wants it so.
+        """
+        parts = [
+            self.product_name,
+            self._displayable_region_code,
+            f"{self.dataset.datetime:%Y-%m-%d}",
+            self.dataset.maturity,
+        ]
+        return "_".join(parts)
+
+    @property
+    def platform_abbreviated(self) -> Optional[str]:
+        # For now from Alchemist the platform is always landsat for C3 processing
+        return "ls"
```

Comment:
This might cause future accidents. Maybe throw an error if platform doesn't contain "landsat"?

Ideally, long term, we move this logic up to the parent function as a new case.  
 - Presumably? "if platform contains multiple landsats, only show 'ls'"